pkg/auth/requests/authenticate.go

```diff
@@ -158,11 +157,30 @@ func (a *tokenAuthenticator) Authenticate(req *http.Request) (*AuthenticatorResp
 	authResp.User = token.UserID
 	authResp.UserPrincipal = token.UserPrincipal.Name
 	authResp.Groups = groups
-	logrus.Debugf("Auth filter calling provider to get extra info %v", token.AuthProvider)
 
-	extraFromProvider := providers.GetUserExtraAttributes(token.AuthProvider, token)
-	if extraFromProvider != nil {
-		authResp.Extras = extraFromProvider
+	if attribs != nil {
+		if strings.EqualFold(token.AuthProvider, "local") || token.AuthProvider == "" {
+			//gather all extraInfo for all external auth providers present in the userAttributes
```

Comment:
For kubectl access via kubeconfig token, the auth provider is always set as "local" in that token. So we gather extraInfo of all external authProviders and log that to audit-log. The user info for "local" auth is already set in the audit logging, what we desire is having the external IDP (IdentityProvider) values. For UI access, the session tokens always have the correct authProvider configured, so only that provider's extra-info will be passed down.